File: java/climbingStairs/ClimbingStairs.java
package de.entwicklerheld.climbingStairs;

import java.util.ArrayList;
import java.util.List;
import java.util.Arrays;

public class ClimbingStairs {

    public static int climbingStairs(int numberOfStairs) {
        if(numberOfStairs < 0) return 0;

        List<Integer> dp = new ArrayList<>(  Arrays.asList(0,1,2));

        for(int i = 3; i <= numberOfStairs; i++) {
            dp.add(dp.get(i-1)+dp.get(i-2));
        }

        return dp.get(numberOfStairs);

    }

}

File: php/climbingStairs/ClimbingStairs.php
<?php
declare(strict_types=1);

function climbingStairs(int $numberOfStairs) {
    if($numberOfStairs <= 0) return 0;

    $dp = [0,1,2];

    for($i = 3; $i <= $numberOfStairs; $i++) {
        $dp[$i] = $dp[$i-1] + $dp[$i-2];
    }

    return $dp[$numberOfStairs];
}